main/src/ca/uwaterloo/flix/language/ast/MonoType.scala

```diff
@@ -66,9 +66,11 @@ object MonoType {
 
   case class Arrow(args: List[MonoType], result: MonoType) extends MonoType
 
-  case class RecordEmpty() extends MonoType
+  case class EmptyRecordRow() extends MonoType
 
-  case class RecordExtend(label: String, value: MonoType, rest: MonoType) extends MonoType
+  case class ExtendedRecordRow(label: String, value: MonoType, rest: MonoType) extends MonoType
+
+  case class Record(row: MonoType) extends MonoType
 
```

Comment:
The MonoTypes should be left unchanged by this PR. I don't think there is a reason to change them-- they are only used in the later stages of the compiler and they are deliberately kept simple.

Reply:
Edit: The monotypes are a simplification of Type that are used in the later stages of the compiler. They are deliberately kept simple.

Reply:
What about for NamedAst.Type?

Reply:
No. I think it is only Type that we want to change.